platforms/ide/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApi.groovy

```diff
@@ -238,10 +237,7 @@ class ToolingApi implements TestRule {
             // When using an isolated user home, first initialise the Gradle instance using the default user home dir
             // This sets some static state that uses files from the user home dir, such as DLLs
             connector.useGradleUserHomeDir(new File(context.gradleUserHomeDir.path))
-
-            try (def connection = connector.connect()) {
-                connection.getModel(BuildEnvironment.class)
-            }
```

Comment:
> does Buildship use this at all?

Yes, Buildship uses `BuildEnviromment` to get the used Gradle version. The code assumes that's it's a side-effect free, cheap operation. In reality it can trigger a Gradle wrapper download, so it's incorrect already. I can look into cleaning up the usage if we decide to change the behaviour and launch the daemon when the tapi model queries the `BuildEnvironment` model.

Reply:
> Do you know what's going on here @donat ? it sort of looks like a sanity check that the connection is "good"?

Looking at the history, it seems that Adam introduced the code to avoid some sort of file locking issues on Windows when the tests were executed in embedded mode. I don't have further knowledge on it.